test/NetTopologySuite.Samples.Console/Technique/SplitPolygonAtDateline.cs

```diff
@@ -0,0 +1,802 @@
+            if (delNeg < delPos)
+                lonSplit = 180d + shift;
+            Debug.Print("  lonSplit={0}", lonSplit);
+
+            // build a series of coordinates
+            var cutterPtList = new List<Coordinate>();
+            cutterPtList.Add(new Coordinate(lonSplit, +100));
+            cutterPtList.Add(new Coordinate(lonSplit, -100));
+
+            // create a polyline from the coordinates
+            var cutterLine = new LineString(cutterPtList.ToArray());
+
+            // we can densify the line but this seems to confuse the splitting routine
+            // cutterLine = (LineString)Densifier.Densify(cutterLine, 0.1); // densify to a specific resolution 0.1 degrees
+
+            // return the "new" dateline polyline that has been shifted
+            return cutterLine;
+        }
+
+        /// <summary>
+        /// Define the cutter polygon that is based on the dateline with a shift to put it in the same hemisphere as the polygon.
+        /// We also use the trim gap as the polygon width.
+        /// </summary>
+        /// <param name="delNeg"></param>
+        /// <param name="delPos"></param>
+        /// <param name="shift"></param>
+        /// <param name="trimGap"></param>
+        /// <returns></returns>
+        private static Polygon DatelineCutterPolygon(double delNeg, double delPos, double shift, double trimGap)
+        {
+            Debug.Print("DatelineCutterPolygon: delNeg={0} delPos={1} shift={2} trimGap={3}", delNeg, delPos, shift, trimGap);
+
+            if (double.IsInfinity(trimGap) || double.IsNaN(trimGap))
+            {
+                throw new Exception("Invalid value for trim gap");
+            }
+
+            double lonSplit = shift - 180d;
+            if (delNeg < delPos)
+                lonSplit = 180d + shift;
+            Debug.Print("  lonSplit={0} lonSplit+trimGap={1}", lonSplit, lonSplit + trimGap);
+
+            // build a clockwise series of coordinates
+            var cutterPtList = new List<Coordinate>();
+            cutterPtList.Add(new Coordinate(lonSplit, -100));
+            cutterPtList.Add(new Coordinate(lonSplit, +100));
+            cutterPtList.Add(new Coordinate(lonSplit + trimGap, +100));
+            cutterPtList.Add(new Coordinate(lonSplit + trimGap, -100));
+            cutterPtList.Add(new Coordinate(lonSplit, -100));
+
+            // create a polygon from the coordinates
+            var pRing = new LinearRing(cutterPtList.ToArray());
+            var cutterPolygon = new Polygon(pRing);
+
+            // we can densify the polygon but this seems to confuse the difference routine
+            // cutterPolygon = (Polygon)Densifier.Densify(cutterPolygon, 0.1); // densify to a specific resolution 0.1 degrees
+
+            // return the "new" dateline polygon that has been shifted and also has a gap from the dateline
+            return cutterPolygon;
+        }
+
+        private static void DebugPrintCoords(List<Coordinate> coords, string hdr)
```

Comment:
Give these `Debug...` functions a return type of `string` use a `StringBuilder` to assemble the output and call the functions using `System.Diagnostics.Debug.WriteLine(~Debug~...)`

Reply:
I will probably take all that out here after Im done with the other suggestions. It was more for me than anything else. Thank you for the feedback.